### PR TITLE
Test flutter app

### PR DESCRIPTION
**Pull request by MarcoMartins2417**

### CODE REVIEW

File: test_flutter_app/integration_test/app_test.dart [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+
+void main() {
+  group('Testing App Performance Tests', () {
+    final binding = IntegrationTestWidgetsFlutterBinding.ensureInitialized()
+        as IntegrationTestWidgetsFlutterBinding;
+
+    binding.framePolicy = LiveTestWidgetsFlutterBindingFramePolicy.fullyLive;
+
+  testWidgets('Scrolling test', (tester) async {
```
Comment (andre-exd) [inline test_flutter_app/integration_test/app_test.dart]: Is this file formatted?

File: test_flutter_app/lib/main.dart [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import 'models/favorites.dart';
+import 'screens/favorites.dart';
+import 'screens/home.dart';
+
+void main() {
+  runApp(TestingApp());
+}
+
+class TestingApp extends StatelessWidget {
+
```
Comment (andre-exd) [inline test_flutter_app/lib/main.dart]: Remove extra line

File: test_flutter_app/lib/models/favorites.dart [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  void add(int itemNo) {
+    _favoriteItems.add(itemNo);
+    notifyListeners();
+  }
+
+  void remove(int itemNo) {
+    _favoriteItems.remove(itemNo);
+    notifyListeners();
+  }
+}
```
Comment (andre-exd) [inline test_flutter_app/lib/models/favorites.dart]: EOF

File: test_flutter_app/test/favorites_test.dart [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:provider/provider.dart';
+import '../lib/models/favorites.dart';
+import '../lib/screens/favorites.dart';
+
+late Favorites favoritesList;
+
+Widget createFavoritesScreen() => ChangeNotifierProvider<Favorites>(
+      create: (context) {
```
Comment (andre-exd) [inline test_flutter_app/test/favorites_test.dart:10]: Fix format

File: test_flutter_app/test/favorites_test.dart [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    });
+
+    testWidgets('Testing Remove Button', (tester) async {
+      await tester.pumpWidget(createFavoritesScreen());
+      addItems();
+      await tester.pumpAndSettle();
+      var totalItems = tester.widgetList(find.byIcon(Icons.close)).length;
+      await tester.tap(find.byIcon(Icons.close).first);
+      await tester.pumpAndSettle();
+      expect(tester.widgetList(find.byIcon(Icons.close)).length,
```
Comment (andre-exd) [inline test_flutter_app/test/favorites_test.dart]: Fix formatting with 120 characters

File: test_flutter_app/test/home_test.dart [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:provider/provider.dart';
+import '../lib/models/favorites.dart';
+import '../lib/screens/home.dart';
+
+Widget createHomeScreen() => ChangeNotifierProvider<Favorites>(
+      create: (context) => Favorites(),
```
Comment (andre-exd) [inline test_flutter_app/test/home_test.dart:8]: Fix formatting

File: test_flutter_app/test/models/favorites_test.dart [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+    test('A new item should be added', () {
+      var number = 45;
+      favorites.add(number);
+      expect(favorites.items.contains(number), true);
+      favorites.remove(number);
+      expect(favorites.items.contains(number), false);
+    });    
+  });
+}
```
Comment (andre-exd) [inline test_flutter_app/test/models/favorites_test.dart]: EOF

**Review by andre-exd — CHANGES_REQUESTED** [review]

File: test_flutter_app/lib/main.dart (hunk 2 above)
Comment (MarcoMartins2417) [reply test_flutter_app/lib/main.dart]: I think that line makes sense because it's used to separate the main method from the class

**Review by andre-exd — APPROVED** [review]

**Pull request merged**